fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CTASAnalyzer.java

```diff
@@ -170,10 +170,12 @@ private Type transformType(Type srcType) {
                 stringType.setAssignedStrLenInColDefinition();
                 newType = stringType;
             } else if (PrimitiveType.FLOAT == srcType.getPrimitiveType() ||
-                    PrimitiveType.DOUBLE == srcType.getPrimitiveType()) {
+                       PrimitiveType.DOUBLE == srcType.getPrimitiveType()) {
                 newType = ScalarType.createDecimalV3Type(PrimitiveType.DECIMAL128, 38, 9);
```

Comment:
Why does DOUBLE type in src corresponds to DECIMALV3 but not DOUBLE?

Reply:
Because Double does not support key columns. CTAS may build a Double for the Key column.